File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] ans=new int[nums.length];
        int[] reversed= new int[nums.length];
        List<Integer> forward=new ArrayList<>();
        List<Integer> back=new ArrayList<>();
        int product=1;
        int reverse=1;
        for(int i=0;i<nums.length;i++){
            reversed[nums.length-i-1]=nums[i];
        }
        for (int num:nums){

            forward.add(product);
            product*=num;
        }
        for (int rev: reversed){
            back.add(reverse);
            reverse*=rev;            
        }
        for (int j=0;j<ans.length;j++){
            ans[j]=forward.get(j)*back.get(back.size()-j-1);
        }
        return ans;


    }
}  

File: Data Structures & Algorithms/jump-game/submission-4.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        nums.reverse()
        left=right=0
        while right<len(nums)-1:
            right+=1
            if right-nums[right]<=left:
                left=right
        print(left)
        if left==len(nums)-1:
            return True
        else:
            return False        